test/common/http/matching/inputs_test.cc

```diff
@@ -4,19 +4,26 @@
 #include "source/common/http/matching/inputs.h"
 #include "source/common/network/address_impl.h"
 #include "source/common/network/socket_impl.h"
+#include "source/common/stream_info/stream_info_impl.h"
 
+#include "test/test_common/test_time.h"
 #include "test/test_common/utility.h"
 
 namespace Envoy {
 namespace Http {
 namespace Matching {
 
-TEST(MatchingData, HttpRequestHeadersDataInput) {
-  HttpRequestHeadersDataInput input("header");
-  Network::ConnectionInfoSetterImpl connection_info_provider(
+// Helper function to create the stream info.
+StreamInfo::StreamInfoImpl createStreamInfo() {
+  auto connection_info_provider = std::make_shared<Network::ConnectionInfoSetterImpl>(
       std::make_shared<Network::Address::Ipv4Instance>(80),
       std::make_shared<Network::Address::Ipv4Instance>(80));
-  HttpMatchingDataImpl data(connection_info_provider);
+  return {Http::Protocol::Http2, Event::GlobalTimeSystem().timeSystem(), connection_info_provider};
```

Comment:
Can you use a constructor? This is error-prone when fields change in a struct.

Reply:
Thanks for review. Done.